applications/MeshingApplication/custom_utilities/mmg_utilities.cpp

```diff
@@ -0,0 +1,2556 @@
+    MMGMeshInfo<TMMGLibrary>& rMMGMeshInfo,
+    const IndexType EchoLevel
+    )
+{
+    rMMGMeshInfo.NumberOfNodes = mMmgMesh->np;
+    if (TMMGLibrary == MMGLibrary::MMG2D) { // 2D
+        rMMGMeshInfo.NumberOfLines = mMmgMesh->na;
+    } else if (TMMGLibrary == MMGLibrary::MMG3D) { // 3D
+        rMMGMeshInfo.NumberOfTriangles = mMmgMesh->nt;
+        rMMGMeshInfo.NumberOfQuadrilaterals = mMmgMesh->nquad;
+    } else { // Surfaces
+        rMMGMeshInfo.NumberOfLines = mMmgMesh->na;
+    }
+    if (TMMGLibrary == MMGLibrary::MMG2D) { // 2D
+        rMMGMeshInfo.NumberOfTriangles = mMmgMesh->nt;
+    } else if (TMMGLibrary == MMGLibrary::MMG3D) { // 3D
+        rMMGMeshInfo.NumberOfTetrahedra = mMmgMesh->ne;
+        rMMGMeshInfo.NumberOfPrism = mMmgMesh->nprism;
+    } else { // Surfaces
+        rMMGMeshInfo.NumberOfTriangles = mMmgMesh->nt;
+    }
+
+    KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) << "\tNodes created: " << rMMGMeshInfo.NumberOfNodes << std::endl;
+    if (TMMGLibrary == MMGLibrary::MMG2D) { // 2D
+        KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) <<
+        "Conditions created: " << rMMGMeshInfo.NumberOfLines << "\n" <<
+        "Elements created: " << rMMGMeshInfo.NumberOfTriangles << std::endl;
+    } else if (TMMGLibrary == MMGLibrary::MMG3D) { // 3D
+        KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) <<
+        "Conditions created: " << rMMGMeshInfo.NumberOfTriangles + rMMGMeshInfo.NumberOfQuadrilaterals << "\n\tTriangles: " << rMMGMeshInfo.NumberOfTriangles << "\tQuadrilaterals: " << rMMGMeshInfo.NumberOfQuadrilaterals << "\n" <<
+        "Elements created: " << rMMGMeshInfo.NumberOfTetrahedra + rMMGMeshInfo.NumberOfPrism << "\n\tTetrahedron: " << rMMGMeshInfo.NumberOfTetrahedra << "\tPrisms: " << rMMGMeshInfo.NumberOfPrism << std::endl;
+    } else { // Surfaces
+        KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) <<
+        "Conditions created: " << rMMGMeshInfo.NumberOfLines << "\n" <<
+        "Elements created: " << rMMGMeshInfo.NumberOfTriangles << std::endl;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<MMGLibrary TMMGLibrary>
+IndexVectorType MmgUtilities<TMMGLibrary>::CheckNodes(
+    ModelPart& rModelPart,
+    const IndexType EchoLevel
+    )
+{
+    DoubleVectorMapType node_map;
+
+    IndexVectorType nodes_to_remove_ids;
+
+    DoubleVectorType coords(Dimension);
+
+    auto& r_nodes_array = rModelPart.Nodes();
+    const auto it_node_begin = r_nodes_array.begin();
+
+    for(SizeType i = 0; i < r_nodes_array.size(); ++i) {
+        auto it_node = it_node_begin + i;
+
+        const array_1d<double, 3>& r_coordinates = it_node->Coordinates();
+
+        for(IndexType i_coord = 0; i_coord < Dimension; i_coord++)
+            coords[i_coord] = r_coordinates[i_coord];
+
+        node_map[coords] += 1;
```

Comment:
There is a performance overhead when you recompute the hash. It is better to hold a reference so you only compute it once.
```suggestion
        auto& r_count = node_map[coords];
        r_count += 1;
        
        if (r_count > 1)
        ...
```